TenureInformationApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -41,5 +42,19 @@ public async Task<TenureInformation> PostNewTenureAsync(CreateTenureRequestObjec
             return tenureDbEntity.ToDomain();
 
         }
+
+        [LogCall]
+        public async Task<TenureInformation> UpdateTenure(TenureQueryRequest query, UpdateTenureRequestObject updateTenureRequest)
+        {
+            _logger.LogDebug($"Calling IDynamoDBContext.LoadAsync for id {query.Id} and then IDynamoDBContext.SaveAsync");
+            var load = await _dynamoDbContext.LoadAsync<TenureInformationDb>(query.Id).ConfigureAwait(false);
+            if (load == null) return null;
```

Comment:
if there is no object to return, should we return null or should we throw an exception? (given this is an update, shouldn't an object always be there? )

Reply:
So the gateway would return null to the controller and then the controller would send a Not Found response back to the client with the id value